sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -218,7 +218,7 @@ class ResolveSessionCatalog(val catalogManager: CatalogManager)
           throw QueryCompilationErrors.commandNotSupportNestedColumnError(
             "DESC TABLE COLUMN", toPrettySQL(child))
         case other =>
-          throw new AnalysisException(s"[BUG] unexpected column expression: $other")
+          throw QueryCompilationErrors.unexpectedColumnExpressionError(other)
```

Comment:
[BUG] indicates that it's an assert-like error. We can simply change the exception type to `IllegalStateException`.

Reply:
OK